plugins/catalog-common/src/search/CatalogEntityDocument.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { IndexableDocument } from '@backstage/search-common';
+
+/**
+ * The Document format for an Entity in the Catalog for search
+ *
+ * @public
+ */
+export interface CatalogEntityDocument extends IndexableDocument {
+  /** @deprecated use `type` as well, as `componentType` will be removed after a few releases but we dont want to break indexing */
```

Comment:
```suggestion
  /** @deprecated `componentType` is being renamed to `type`. During the transition both of these fields should be set to the same value, in order to avoid issues with indexing. */
```